server/series/api.py

```diff
@@ -0,0 +1,461 @@
+        return 400, {"message": "Series/Team does not exist"}
+
+    if request.user not in team.admins.all():
+        return 401, {"message": "Only team admins can register a team to a series !"}
+
+    # Any team can register for the Club series
+    # School, College, State teams can register for the appropriate series only
+    if (
+        series.category != Series.Category.CLUB  # noqa: PLR1714
+        and series.category != team.category
+    ):
+        return 400, {"message": f"Only {series.category} teams can register for this series"}
+    else:
+        series.teams.add(team)
+
+    return 200, series
+
+
+@router.delete(
+    "/{series_slug}/deregister-team", response={200: SeriesSchema, 400: Response, 401: Response}
+)
+def remove_team_series_registration(
+    request: AuthenticatedHttpRequest,
+    series_slug: str,
+    team_details: AddOrRemoveTeamSeriesRegistrationSchema,
+) -> tuple[int, Series | message_response]:
+    try:
+        series = Series.objects.get(slug=series_slug)
+        team = Team.objects.get(slug=team_details.team_slug)
+    except (Series.DoesNotExist, Team.DoesNotExist):
+        return 400, {"message": "Series/Team does not exist"}
+
+    if request.user not in team.admins.all():
+        return 401, {"message": "Only team admins can de-register from team a series !"}
+
+    series.teams.remove(team)
+
+    return 200, series
+
+
+@router.get("/{series_slug}/team/{team_slug}", response={200: TeamSchema, 400: Response})
+def get_series_team(
+    request: AuthenticatedHttpRequest, series_slug: str, team_slug: str
+) -> tuple[int, Team | message_response]:
+    try:
+        series = Series.objects.get(slug=series_slug)
+    except Series.DoesNotExist:
+        return 400, {"message": "Series does not exist"}
+
+    try:
+        team = Team.objects.get(slug=team_slug)
+    except Team.DoesNotExist:
+        return 400, {"message": "Team does not exist"}
+
+    if team not in series.teams.all():
+        return 400, {"message": f"{team.name} is not registered for {series.name} !"}
+
+    return 200, team
+
+
+# @router.get(
+#     "/{series_slug}/team/{team_slug}/players/search",
+#     response={200: list[PlayerTinySchema]},
+# )
+# @paginate(PageNumberPagination, page_size=5)
+# def search_players(
+#     request: AuthenticatedHttpRequest, series_slug: str, team_slug: str, text: str = ""
+# ) -> list[Player] | QuerySet[Player]:
+#     try:
+#         series = Series.objects.get(slug=series_slug)
+#     except Series.DoesNotExist:
+#         return []
+
+#     try:
+#         team = Team.objects.get(slug=team_slug)
+#     except Team.DoesNotExist:
+#         return []
+
+#     return series_team_players_search_list(
+#         series=series, team=team, search_text=text.strip().lower()
+#     )
+
+
+@router.post(
+    "/{series_slug}/team/{team_slug}/invitation",
+    response={200: SeriesRosterInvitationCreateSchema, 400: Response, 401: Response},
+)
+def send_series_invitation(
```

Comment:
we need to check if this player does not have any other registration with other team for the same series before sending the invitation

Reply:
done